spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/LettuceConnectionConfiguration.java

```diff
@@ -82,6 +85,11 @@ LettuceConnectionFactory redisConnectionFactory(
 	}
 
 	private LettuceConnectionFactory createLettuceConnectionFactory(LettuceClientConfiguration clientConfiguration) {
+		if (StringUtils.hasText(this.getProperties().getUrl())) {
```

Comment:
if (StringUtils.hasText(getProperties().getUrl())) {

Reply:
`Checkstyle` nor `format` seem to mind but it is not needed and inconsistent w/ the rest of the `getProperties` invocations in here.

Reply:
I am not sure about this one. The behavior is that we're checking for specific properties for `.cluster` or `.sentinel` and configure that if present. This reverse the trend if an URI is set, only for Lettuce.

This makes it quite hard to resonate about. Also, any of those properties are ignored and fully replaced by the URI, which sounds like quite a big change.

Reply:
> This reverse the trend if an URI is set, only for Lettuce.

True. Now the order they get connection config is:
- Jedis: `sentinel` -> `cluster` -> `standalone (which then uses uri)`
- Lettuce: `url` -> `sentinel` -> `cluster` -> `standalone (which then uses uri)`

This does create divergence between the 2 providers. Is the difference between the 2 providers the main concern, or the change in Lettuce connection config behavior? 

> Also, any of those properties are ignored and fully replaced by the URI, which sounds like quite a big change

Yeh this is why I was initially thinking about adding similar validation to [MongoPropertiesClientSettingsBuilderCustomizer](https://github.com/spring-projects/spring-boot/blob/3e21ae12dc2653ab7ce7d561c29bc018a57a4103/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mongo/MongoPropertiesClientSettingsBuilderCustomizer.java#L59) - which would force the consumer to use either the url or the direct props.

Reply:
>  Is the difference between the 2 providers the main concern, or the change in Lettuce connection config behavior?

Both.